lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -331,6 +331,48 @@ ::mlir::LogicalResult mlir::tt::ttir::Conv2dOp::verify() {
   return success();
 }
 
+// Common verifier for all Quantize ops.
+::mlir::LogicalResult
+verifyQuantizeOpCommon(::mlir::Operation *op,
+                       ::mlir::RankedTensorType inputType,
+                       ::mlir::RankedTensorType outputType) {
+  // Sanity check to make sure that input rank matches the rank of the output
+  // tensor.
+  if (inputType.getRank() != outputType.getRank()) {
+    return op->emitOpError() << "Input tensor rank of " << inputType.getRank()
+                             << " does not match the output tensor rank of "
+                             << outputType.getRank();
+  }
+
+  // Shapes of input and output of a quantize operation must be the same.
+  if (inputType.getShape() != outputType.getShape()) {
+    return op->emitOpError("Input and output shapes must be the same.");
+  }
+
+  return ::mlir::success();
+}
+
+// QuantizeOp verification.
+::mlir::LogicalResult mlir::tt::ttir::QuantizeOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  return verifyQuantizeOpCommon(*this, inputType, outputType);
```

Comment:
We can simplify it for all the verify functions as
```suggestion
  return verifyQuantizeOpCommon(*this, getInput().getType(), getOutput().getType());
```

Reply:
Thanks. Resolved in latest commit.